### PR TITLE
feature: include a handy script in the snap for self testing

### PR DESCRIPTION
**Pull request by mattkae**
## What's new?
- Added the `selftest` script which will run all tests back-to-back
- Mentioned this script in the `README`

### CODE REVIEW

File: selftest/selftest [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+fi
+
+if [ -v DISPLAY ]
+then
+    snap run mir-test-tools.gtk3-test
+    snap run mir-test-tools.qt-test
+    snap run mir-test-tools.sdl2-test
+    snap run mir-test-tools.smoke-test
+    snap run mir-test-tools.performance-test
+    snap run mir-test-tools.mir-flutter-app
```
Comment (AlanGriffiths) [inline selftest/selftest]: There's no such app

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
I guess we can improve the script separately and treat it's lack of documentation and general flakiness as "preexisting".

But failing because one of the apps isn't there needs fixing

File: selftest/selftest (hunk 1 above)
Comment (mattkae) [reply selftest/selftest]: Whoops! Meant `mir-test-tools.flutter-app`

Comment (AlanGriffiths) [reply selftest/selftest]: And... that just hangs with "Watches established."

Comment (AlanGriffiths) [reply selftest/selftest]: .. I don't think that "app" is a test

Comment (mattkae) [reply selftest/selftest]: ....

Comment (mattkae) [reply selftest/selftest]: You are right!

Comment (mattkae) [reply selftest/selftest]: Fixed :+1:

**Review by AlanGriffiths — APPROVED** [review]
There are some issues, noted for posterity:

0. The progress and error reporting leave a lot to be desired
1. This script won't work on a VT without first connecting logind-session-control
2. The sdl2 tests fail and abort the script
3. This script can't work on mir:wayland
4. This script won't work on mir:x11 without connecting x11 (which will cause the performance tests fail)
5. The "only with DISPLAY" tests could also be run hosted by Wayland
6. I think the "sudo..." tests are intended to use VTConsoleServices, but that doesn't actually work out

We should probably request autoconnect for logind-session-control, as that is the most functional path.

Approving, as all of this is "pre-existing"

**Pull request merged**